### PR TITLE
Fixed the HWC crash due to improper thread initialization sequence.

### PR DESCRIPTION
**Pull request by shuangwan01**
The thread of GpuDevice is started before DRM Display Manager's Initialization method, there is a chance that GpuDevice thread call method of display manager but underlying DRM display manager implementation still not fully initialized yet. This will lead HWC crash.

JIRA: OAM-63288
Test: Switch clone & mosaic mode and no crash during boot.

### CODE REVIEW

File: common/core/gpudevice.cpp [hunk 1]
```diff
@@ -64,6 +59,10 @@ bool GpuDevice::Initialize() {
   HandleHWCSettings();
   InitializeHotPlugEvents();
 
+  if (!InitWorker()) {
```
Comment (kalyankondapally) [inline common/core/gpudevice.cpp]: shouldn't be needed. DisplayManager is now created before DisplayManager is created.

Comment (kalyankondapally) [reply common/core/gpudevice.cpp]: check with latest code.

Comment (shuangwan01) [reply common/core/gpudevice.cpp]: Yes, DisplayManager is created but not fully initialized. If InitWorker called just next to DisplayManager creation, thread GpuDevice will be put into running status. It has a chance to call methods of display manager but the later one might be not fully initialized yet. The consequence of it is HWC might be crash unexpectedly.

Comment (kalyankondapally) [reply common/core/gpudevice.cpp]: in this case initializing thread doesn't make sense if lock_fd_ == -1 and also in HandleRoutine u can remove call to HandleHWCSettings

Comment (kalyankondapally) [reply common/core/gpudevice.cpp]: Also u can remove initialization_state_lock_ in  InitializeHotPlugEvents and HandleHWCSettings

Comment (shuangwan01) [reply common/core/gpudevice.cpp]: For commit afa1e13, the GpuDevice thread initialization sequence works for me, thanks

**Pull request closed without merging**